### PR TITLE
#105 contact display dialog route state

### LINKED ISSUE #105: [TASK][display contact dialog] the route state for the contact dialog
[STORY](https://app.zenhub.com/workspaces/ft---calendar--contacts-dev-5ff3db9704b1b3001295e51f/issues/openpaas-suite/esn-frontend-contacts/92)

TODO:
* [ ] update the existing route state ( `contact.addressbooks.show` create a new route state ( ie `/show/:adressbookId/contacts/:contactId` )
* [ ] inject the settings component with the `$modal` into the scope when entering the state
* [ ] hide the dialog when leaving the state
* [ ] pass the previous state as params to the state
* [ ] load the contact using the addressbookId and contactId from the $stateParams in the component controller ( ie from $stateParams.contactId and $stateParams.addressbookId )
* [ ] make sure that the contact dialog can be accessed via a URL
* [ ] make it that when you click on the contact you go to the new state `$state.go('example')`:
    - you can use the ContactLocationHelper.show or update it or remove it ( depends on your choice )

### PR DESCRIPTION
**Pull request by rezk2ll**
Resolves #105.

demo using a dummy template

https://user-images.githubusercontent.com/6764881/111955573-f6f20000-8ae9-11eb-9180-47244db39d4e.mp4

### CODE REVIEW

**Comment by huy-ta** [comment]
Mhhhh... splitting tasks like this makes the feature incomplete. Merging this means breaking the whole feature.

**Comment by rezk2ll** [comment]
> Mhhhh... splitting tasks like this makes the feature incomplete. Merging this means breaking the whole feature.

ye that's why i created a feature branch `contact-display` for this. i'm not mergin against `main`

also splitting the route state tasks and template tasks was not a good idea :laughing:

**Review by huy-ta — APPROVED** [review]

File: src/linagora.esn.contact/app/app.routes.js [hunk 1]
```diff
@@ -82,15 +82,23 @@ require('./services/contact-configuration.service.js');
...
-                user: routeResolver.session('user')
-              }
+          params: {
+            previousState: 'contact.addressbooks'
+          },
+          resolve: {
+            modalInstance: function($modal) {
+              return $modal({
+                template: '<contact-show />',
+                keyboard: false,
```
Comment (huy-ta) [inline src/linagora.esn.contact/app/app.routes.js:92]: Btw what does this option do?

**Comment by huy-ta** [comment]
The demo is not what the code in this PR can do, is it? I've tried to duplicate the code for the "Edit contact" issue: https://github.com/OpenPaaS-Suite/esn-frontend-contacts/issues/110, but it doesn't work as expected.

**Comment by rezk2ll** [comment]
> The demo is not what the code in this PR can do, is it? I've tried to duplicate the code for the "Edit contact" issue: #110, but it doesn't work as expected.

yes, because you actually need the template to see the demo ( i didn't include a dummy template into my pr )

File: src/linagora.esn.contact/app/app.routes.js (hunk 1 above)
Comment (rezk2ll) [reply src/linagora.esn.contact/app/app.routes.js:92]: prevent closing the modal using the keyboard

**Comment by huy-ta** [comment]
> > The demo is not what the code in this PR can do, is it? I've tried to duplicate the code for the "Edit contact" issue: #110, but it doesn't work as expected.
> 
> yes, because you actually need the template to see the demo ( i didn't include a dummy template into my pr )

OK.

File: src/linagora.esn.contact/app/app.routes.js (hunk 1 above)
Comment (huy-ta) [reply src/linagora.esn.contact/app/app.routes.js:92]: That's not nice for users without a mouse :jack_o_lantern:.

**Pull request merged**